dotnet/src/SemanticKernel.Core/Functions/SemanticFunction.cs

```diff
@@ -137,6 +108,67 @@ public string ToString(bool writeIndented)
         this._view = new(() => new(functionName, pluginName, promptTemplateConfig.Description, this.Parameters));
     }
 
+    /// <summary>
+    /// Create a semantic function instance, given a semantic function configuration.
+    /// </summary>
+    /// <param name="pluginName">Name of the plugin to which the function being created belongs.</param>
+    /// <param name="functionName">Name of the function to create.</param>
+    /// <param name="promptTemplateConfig">Prompt template configuration.</param>
+    /// <param name="promptTemplate">Prompt template.</param>
+    /// <param name="loggerFactory">The <see cref="ILoggerFactory"/> to use for logging. If null, no logging will be performed.</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <returns>SK function instance.</returns>
+    public static ISKFunction FromSemanticConfig(
```

Comment:
If we are not sticking with Semantic Function naming, I would avoid keeping it in any new public API's we create/modify.

- FromPromptConfig 
- FromPromptModel
- FromConfig
- FromModel